packages/db/src/models/base.ts

```diff
@@ -96,6 +96,25 @@ class BasePgModel<ItemType, RecordType extends { cumulus_id: number }> {
     return record.cumulus_id;
   }
 
+  /**
+   * Get cumulus_id column value for multiple records in Postgres
+   *
+   * @param {Knex | Knex.Transaction} knexOrTransaction -
+   *  DB client or transaction
+   * @param {Function} whereFunction -
+   *  An object or any portion of an object of type RecordType
+   * @returns {Promise<Array<number>>} An array of cumulus_ids for the returned records
+   */
+  async getRecordsCumulusIds(
+    knexOrTransaction: Knex | Knex.Transaction,
+    whereFunction: Function
```

Comment:
I don't know if I understand the reason for using a function as the argument type vs an object or array or whatever. It's a little more than a nitpick because a function argument can't be typed very easily to prevent misuse

Reply:
This was me trying to make a whereIn query work while having the same basic params as the single record version of this function since there's no way to pass it an object https://knexjs.org/#Builder-whereIn

However, I've now changed it so that we just pass in the columnNames and the values

Reply:
Agree, adding a comment here in a few.

Reply:
So the reason for using the function is to expose the .whereIn method via the builder, and *that* is needed because we always await results and don't expose the raw knex object back. 

A couple of options:

Reply:
Off the cuff and hastily reassembled: 

I think the current stance was to attempt to avoid generic abstraction that isn't type safe by making the query logic explicit in a /lib helper, particularly in the feature branch.  That may be the path of least resistance, but perhaps inelegant.   

We could opt to return the search knex object without awaiting it, allowing users to chain.   That would allow for queries like 

`await granulePgModel.search(knex, {}).whereIn('granule_id', ['foo'])`
`await granulePgModel.search(knex, {}).where((builder) => { granuleId: 'foo'})`

of course, that doesn't fix that chaining the .whereIn query like that isn't handled well via return typing, and in some ways just punts the problem to the endpoint logic (and the less logic of that sort in the API the better, given it's lack of typing).   That seems not great. 



Or...

 punt entirely and just expose a base method like 

const query(knexOrTransaction: Knex | Knex.Transaction) { return knexOrTransaction(this.tableName))

Or

We could improve the base model search to allow for something like: 

```
 search(
    knexOrTransaction: Knex | Knex.Transaction,
    params: Partial<RecordType>,
    select: keyof RecordType[],
    whereIn: Partial<RecordType>[]

  ): Promise<RecordType[]> {
    return knexOrTransaction(this.tableName).select(select).where((builder) => {
      builder.where(params);
      if(whereIn) {
        // add lots of wherein clauses or more cleverly type whereIn and do <logic here to support the need> 
      }
.... <snip>
```

that is a bit more charming in that the whereIn could be properly typed both in our method and relative to the query logic. 

:thinking face: